### PR TITLE
POM-757 change caseload route to include staff_id

### PR DESCRIPTION
**Pull request by Katy600**
This pull request changes the caseload route to include staff_id

### CODE REVIEW

File: app/controllers/caseload_controller.rb [hunk 1]
```diff
@@ -84,13 +83,21 @@ def filter_allocations(allocations)
...
-    @pom = PrisonOffenderManagerService.get_signed_in_pom_details(
-      current_user,
-      active_prison_id
-    )
+    @pom = StaffMember.new(params[:staff_id])
+  end
+
+  def ensure_signed_in_pom_is_this_pom
+    user = Nomis::Elite2::UserApi.user_details(current_user)
+    unless staff_id == user.staff_id || current_user_is_spo?
```
Comment (starswan) [inline app/controllers/caseload_controller.rb:91]: think you forgot to check that we are actually a POM?

File: spec/services/allocation_service_spec.rb [hunk 2]
```diff
@@ -47,7 +47,7 @@
             "nomis_offender_id" => "G4273GI",
             "pom_email" => "pom@digital.justice.gov.uk",
             "pom_name" => "Moic",
-            "url" => "http://localhost:3000/prisons/LEI/caseload"
+            "url" => "http://localhost:3000/prisons/LEI/staff/485926/caseload"
```
Comment (starswan) [inline spec/services/allocation_service_spec.rb:50]: would be nice to show that this is the correct POM (primary) and secondary below

File: spec/services/email_service_spec.rb [hunk 3]
```diff
@@ -152,7 +152,7 @@
         offender_name: "Ahmonis, Imanjah",
         offender_no: "G2911GD",
         prison: 'HMP Leeds',
-        url: 'http://localhost:3000/prisons/LEI/caseload'
```
Comment (starswan) [inline spec/services/email_service_spec.rb:155]: ditto here - it's not obvious that we're headed for the correct place

File: app/controllers/caseload_controller.rb (hunk 1 above)
Comment (Katy600) [reply app/controllers/caseload_controller.rb:91]: @starswan  Oh that is a good point. I think I had assumed that that the signed in staff member could only be a pom or an spo.

**Pull request merged**